crates/ruff_linter/src/noqa.rs

```diff
@@ -816,7 +816,7 @@ fn build_noqa_edits_by_line<'a>(
             offset,
             matches
```

Comment:
Can you double check if we need to use a `HashSet` for `noqa_codes` in case there are multiple matches for the same code? Or does it not matter, or does some other code guarantee that there are no duplicates?